### DISCUSSION TITLE
[Stripe paymentIntent] Add Stripe Connect parameters

### DISCUSSION THREAD

**Discussion by benjamin-carriou** [saleor/saleor#11046] [Q&A]
Hi :)

I'am trying to use Saleor with Stripe and it's works.
I can create a `paymentIntent`, and confirm it with a `paymentMethod` and cie.

Now, because my app is a marketplace, I need to capture application fee with Stripe Connect when a payment is make.

In the documentation of Stripe, i need to configure some parameters:

- `application_fee_amount` (more info [here](https://stripe.com/docs/api/payment_intents/create#create_payment_intent-application_fee_amount)) to set amount of fee
- `transfer_data.destination` (more info [here](https://stripe.com/docs/api/payment_intents/create#create_payment_intent-transfer_data)) to set the Connect account ID

My question is the following:

With Saleor, it is possible when a checkout is created, to add theses parameters ?
Or an another possibility ?

Thx in advance for your help :)

**Comment by peelar** [comment]
Hello @benjamin-carriou 👋 
> With Saleor, it is possible when a checkout is created, to add theses parameters ?

Where would you like to add them? Have you looked into [Metadata](https://docs.saleor.io/docs/3.x/developer/metadata)?

**Reply by benjamin-carriou** [reply]
Hi @peelar :)

Metadata are interpreted by Stripe as parameters ?

When I read the Saleor doc, I thought I would follow [these steps](https://docs.saleor.io/docs/3.x/developer/checkout#case-processing-payment-before-creating-an-order)

And so do not go through the paymentIntent created by Saleor but create myself a paymentIntent with these parameters that are fine, then when the payment is in the right status, create an order from the checkout in Saleor.

The problem with paymentIntent in Stripe (from what I understood) is that you can't update them afterwards to add these parameters related to Connect, this is only done when creating the paymentIntent.

**Reply by peelar** [reply]
Can you please have a look at Stripe implementation in [react-storefront](https://github.com/saleor/react-storefront)? Although I don't see any mentions of `transfer_data` or `application_fee_amount` in the codebase, maybe you will find your answers some other way. 

If not, I will pass it on to one of our devs.
